### PR TITLE
Refactor AppController

### PR DESCRIPTION
**Pull request by typfel**
Extract the app state machine into a separate class for easier readbility/testing
and move objects into the ZClientViewController since they are only relevant when
we have a user session.

depends on:
- [x] new SE release

### CODE REVIEW

File: Wire-iOS/Sources/AppState.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    case unauthenticated(error : Error?)
+    case blacklisted
+    case migrating
+    case suspended
+
+    public static func ==(lhs: AppState, rhs: AppState) -> Bool {
+        
+        switch (lhs, rhs) {
+        case (.headless, .headless):
+            fallthrough
```
Comment (vytis) [inline Wire-iOS/Sources/AppState.swift]: `fallthrough` police  🚔

Comment (typfel) [inline Wire-iOS/Sources/AppState.swift]: In this case I think it looks for readably since otherwise it's all on long line.

**Comment by typfel** [comment]
Forgot I did some small SE changes

File: Wire-iOS/Sources/AppState.swift (hunk 1 above)
Comment (vytis) [inline Wire-iOS/Sources/AppState.swift]: You can stack cases:
```
case (.headless, .headless)
case (.authenticated, .authenticated):
...
```

Comment (typfel) [inline Wire-iOS/Sources/AppState.swift]: Oh :)

**Pull request merged**